android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PwoDiscoveryService.java

```diff
@@ -379,4 +424,19 @@ private PendingIntent createReturnToAppPendingIntent() {
     PendingIntent pendingIntent = PendingIntent.getActivity(this, requestID, intent, 0);
     return pendingIntent;
   }
+
+  public void requestPwoMetadata(PwoResponseCallback pwoResponseCallback) {
+    for (PwoMetadata pwoMetadata : mUrlToPwoMetadata.values()) {
+      pwoResponseCallback.onPwoDiscovered(pwoMetadata);
+    }
+    mPwoResponseCallbacks.add(pwoResponseCallback);
```

Comment:
The client code is responsible for cleaning that up.  If it wants to request twice it can.